persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -293,8 +306,42 @@ private void authorizeByToken(ByteBuffer token, CQLStatement statement) {
     }
   }
 
+  private AuthenticationSubject loadAuthenticationSubject(ByteBuffer token) {
+    AuthenticationSubject authenticationSubject;
+    ObjectInput in = null;
+    try {
+      if (token.position() == token.limit()) {
+        token.flip();
+      }
+      byte[] bytes = new byte[token.remaining()];
+      token.get(bytes);
+      ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
+      in = new ObjectInputStream(bis);
+      AuthenticatedUser authenticatedUser = (AuthenticatedUser) in.readObject();
```

Comment:
Java serialization can be slow (or used to be?).. since this in on the critical CQL query path, would it be worth using a custom serialization for the properties of `AuthenticatedUser`?... a bit more work, but hopefully more performant... WDYT?

Reply:
Not sure I follow regarding the custom serialization.

Reply:
I mean constructing a custom byte array from the properties of `AuthenticatedUser` instead of using `ObjectOutputStream` / `ObjectInputStream`.

Reply:
here, specifically, we'd do something like
```
int size = token.getInt()
byte[] chars = new byte[size];
token.get(chars);
String roleName = new String(chars, UTF8);
```

Reply:
for that matter it might be more convenience to put individual `AuthenticatedUser` as separate `ByteBuffers` into the custom payload... makes the deserialization logic easier, I guess.